apps/managedidentity/managedidentity_with_logger_test.go

```diff
@@ -0,0 +1,128 @@
+//go:build go1.21
+
+package managedidentity
+
+import (
+	"bytes"
+	"context"
+	"log/slog"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/internal/mock"
+)
+
+// Custom log handler to capture log output
+type BufferHandler struct {
+	buf bytes.Buffer
+}
+
+// Custom log handler to capture log output and filter out info level logs
+type FilteredBufferHandler struct {
+	buf bytes.Buffer
+}
+
+func (h *BufferHandler) Enabled(ctx context.Context, level slog.Level) bool { return true }
+func (h *BufferHandler) Handle(ctx context.Context, record slog.Record) error {
+	h.buf.WriteString(record.Message + " ")
+	record.Attrs(func(attr slog.Attr) bool {
+		h.buf.WriteString(attr.Key + "=" + attr.Value.String() + " ")
+		return true
+	})
+	return nil
+}
+func (h *BufferHandler) WithAttrs(attrs []slog.Attr) slog.Handler { return h }
+func (h *BufferHandler) WithGroup(name string) slog.Handler       { return h }
+func (h *FilteredBufferHandler) Enabled(ctx context.Context, level slog.Level) bool {
+	return level == slog.LevelDebug
+}
+func (h *FilteredBufferHandler) Handle(ctx context.Context, record slog.Record) error {
+	h.buf.WriteString(record.Message + " ")
+	record.Attrs(func(attr slog.Attr) bool {
+		h.buf.WriteString(attr.Key + "=" + attr.Value.String() + " ")
+		return true
+	})
+	return nil
+}
+func (h *FilteredBufferHandler) WithAttrs(attrs []slog.Attr) slog.Handler { return h }
+func (h *FilteredBufferHandler) WithGroup(name string) slog.Handler       { return h }
+
+func TestClientLogging(t *testing.T) {
+	// Set up mock client
```

Comment:
please forgive nitpicking: I'd remove these comments because the code clearly says what it does